app/infrastructure/database/repositories.py

```
@@ -0,0 +1,265 @@
+
+    async def update_token_usage(self, user_id: str, tokens_used: int) -> None:
+        try:
+            user = await User.get(user_id=user_id)
+            user.token_used += tokens_used
+            await user.save()
+            logger.info(f"Updated token usage for user {user_id}: +{tokens_used}")
+        except Exception as e:
+            logger.error(f"Error updating token usage for user {user_id}: {str(e)}")
+            raise DatabaseError(f"Failed to update token usage: {str(e)}")
+
+    async def create_user(self, user_data: dict) -> User:
```

Ruff (0.11.9)

`User` may be undefined, or defined from star imports

(F405)